c10/core/impl/ThreadLocalState.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <c10/core/DispatchKeySet.h>
+#include <c10/macros/Macros.h>
+
+namespace c10 {
+namespace impl {
+
+// NB: POD, must be zero initialized!
+// Note [TLS Initialization]
+// We would prefer fields to be initialized with non-zero state
+// e.g. BackendSelect and ADInplaceOrView in included set.  But certain Windows
+// compiler (e.g the one used in ARVR tests) only allow TLS to be
+// zero-initialized. To preserve the invariant that raw TLS storage of the
+// default state is zero, we obtain the actual include keyset by XORing
+// the raw bits (included / excluded) with an XOR mask value.
+//
+// Moreover, PODLocalState is a simple data container. Logic for interpreting
+// or updating fields should be owned by higher level containers which operate
+// on a PODLocalState* pointer.
```

Comment:
Since it's hard coded here what PODLocalState actually contains, there is only a fixed number of higher level containers. Help the reader out by giving direct code pointers to the full set of higher level containers.